ydb/docs/ru/core/reference/ydb-sdk/coordination.md

```diff
@@ -0,0 +1,306 @@
+
+{% list tabs %}
+
+- Go
+
+    ```go
+    lease, err := session.AcquireSemaphore(ctx,
+        "mySemaphore",  // semaphore name
+        5,              // value to increase semaphore by
+    )
+    ```
+
+    Для отмены ожидания взятия семафора, достаточно отменить переданный в метод контекст `ctx`.
+
+- C++
+
+    ```cpp
+    session
+        .AcquireSemaphore(
+            "my-semaphore",                       // semaphore name
+            TAcquireSemaphoreSettings().Count(5)  // value to increase semaphore by
+        )
+        .ExtractValueSync()
+        .ExtractResult();
+    ```
+
+    При захвате можно опционально передать структуру `TAcquireSemaphoreSettings` со следующими настройками:
+
+    - `Count` - значение, на которое увеличивается семафор при захвате.
+    - `Data` - дополнительные данные, которые можно положить в семафор.
+    - `OnAccepted` - вызывается, когда операция встаёт в очередь (например, если семафор невозможно было захватить сразу).
+
+      - Не будет вызвано, если семафор захватывается сразу.
+      - Важно учитывать, что вызов может произойти параллельно с результатом `TFuture`.
+
+    - `Timeout` - максимальное время, в течение которого операция может пролежать в очереди на сервере.
+
+      - Операция вернёт `false`, если за время `Timeout` после добавления в очередь не удалось захватить семафор.
+      - При `Timeout` установленном в 0 операция по смыслу работает как `TryAcquire`, т.е. семафор будет либо захвачен атомарно и операция вернёт `true`, либо операция вернёт `false` без использования очередей.
+
+    - `Ephemeral` - если `true`, то имя является эфемерным семафором, такие семафоры автоматически создаются при первом `Acquire` и автоматически удаляются с последним `Release`.
+    - `Shared()` - алиас для выставления `Count = 1`, захват семафора в shared режиме.
+    - `Exclusive()` - алиас для выставления `Count = max`, захват семафора в exclusive режиме (для семафоров, созданных с лимитом `Max<ui64>()`).
+
+{% endlist %}
+
+Взятое значение захваченного семафора можно снизить (но не увеличить), вновь вызвав для него метод `AcquireSemaphore` с меньшим значением.
+
+### Обновление данных семафора {#update-semaphore}
+
+С помощью метода `UpdateSemaphore` можно обновить (заменить) данные семафора, которые были привязаны при его создании.
+
+{% list tabs %}
+
+- C++
+
+    ```cpp
+    session
+        .UpdateSemaphore(
+            "my-semaphore",  // semaphore name
+            "updated-data"   // new semaphore data
+        )
+        .ExtractValueSync()
+        .ExtractResult();
+    ```
+
+{% endlist %}
+
+Этот вызов не требует захвата семафора и не приводит к нему. Если требуется, чтобы данные обновлял только один конкретный клиент, то это необходимо явным образом обеспечить, например, захватив семафор, обновив данные и отпустив семафор обратно.
+
+### Получение данных семафора {#describe-semaphore}
+
+{% list tabs %}
+
+- C++
```

Comment:
```suggestion
- Go

    ```go
    description, err := session.DescribeSemaphore(
        "my-semaphore"                                // semaphore name
        options.WithDescribeOwners(true), // to get list of owners
        options.WithDescribeWaiters(true), // to get list of waiters
    )
    ```

- C++
```